tensorflow/python/estimator/inputs/numpy_io_test.py

```diff
@@ -285,6 +285,77 @@ def testNumpyInputFnWithMismatchLengthOfInputs(self):
             num_epochs=1)
         failing_input_fn()
 
+  def testNumpyInputFnWhenLabelIsDictionary(self):
+    a = np.arange(4) * 1.0
+    b = np.arange(32, 36)
+    x = {'a': a, 'b': b}
+    y = {'y1': np.arange(-32, -28), 'y2': np.arange(32, 28, -1)}
+
+    with self.test_session() as session:
+      input_fn = numpy_io.numpy_input_fn(
+        x, y, batch_size=2, shuffle=False, num_epochs=1)
+      features_tensor, targets_tensor = input_fn()
+
+      coord = coordinator.Coordinator()
+      threads = queue_runner_impl.start_queue_runners(session, coord=coord)
+
+      features, targets = session.run([features_tensor, targets_tensor])
+      self.assertEqual(len(features), 2)
+      self.assertAllEqual(features['a'], [0, 1])
+      self.assertAllEqual(features['b'], [32, 33])
+      self.assertEqual(len(targets), 2)
+      self.assertAllEqual(targets['y1'], [-32, -31])
+      self.assertAllEqual(targets['y2'], [32, 31])
+
+      session.run([features_tensor, targets_tensor])
+      with self.assertRaises(errors.OutOfRangeError):
+        session.run([features_tensor, targets_tensor])
+
+      coord.request_stop()
+      coord.join(threads)
+
+  def testNumpyInputFnWhenLabelIsEmptyDictionary(self):
+    a = np.arange(4) * 1.0
+    b = np.arange(32, 36)
+    x = {'a': a, 'b': b}
+    y = {}
+
+    with self.test_session() as session:
+      input_fn = numpy_io.numpy_input_fn(
+        x, y, batch_size=2, shuffle=False, num_epochs=1)
+      features_tensor = input_fn()
```

Comment:
Is this the right behavior? Or would we expect the input function generated to return an empty labels dict?

Reply:
I have no idea. `y={}` seems a little weird. Do you prefer to returning a empty labels dict?

Reply:
By the way,  is `x = {}` or `y = {}` a valid argument?

Reply:
`x = {}` makes no sense (testing for it and raise an appropriate error would be nice), `y = {}` does make sense for unlabeled problems. As does `y=None`.

However, I got confused because even though the docs don't mention it, this function sometimes returns only `features`, and sometimes it returns `(features, labels)`. 

I would find it somewhat odd if the return type changed based on the content of the dict (or non-content).

Reply:
I agree that we'd better return consistent result `(features, labels)`.  How about this:
+ When x = {}, raise an ValueError
+ When y = None, labels = None
+ When y = {}, labels = {}

Reply:
That's perfect.

Reply:
Oh, I forget to remind you that the compatibility will be broken if we return `None` when `y=None`.